scala/lambdas/ingest-flow-control/src/test/scala/uk/gov/nationalarchives/ingestflowcontrol/Helpers.scala

```diff
@@ -89,15 +89,19 @@ object Helpers {
     override def writeItems[T](tableName: String, items: List[T])(using format: DynamoFormat[T]): IO[List[BatchWriteItemResponse]] = notImplemented
 
     override def queryItems[U](tableName: String, requestCondition: RequestCondition, potentialGsiName: Option[String])(using returnTypeFormat: DynamoFormat[U]): IO[List[U]] =
-      notImplemented
-
-    override def getItems[T, K](primaryKeys: List[K], tableName: String)(using returnFormat: DynamoFormat[T], keyFormat: DynamoFormat[K]): IO[List[T]] =
-      val firstPK = primaryKeys.head.asInstanceOf[IngestQueuePartitionKey]
-      errors.raise(_.getItem, "Error getting item from dynamo table") >>
+      errors.raise(_.queryItem, "Error getting item from dynamo table") >>
         ref.get.map { existing =>
-          existing.filter(_.sourceSystem == firstPK.sourceSystem).sortBy(_.queuedAt).map(_.asInstanceOf[T])
+          (for {
+            values <- Option(requestCondition.attributes.values)
+            map <- values.toMap[String].toOption
+          } yield existing
```

Comment:
1. Maybe `existingItems`?
2. What is `existing`? Items that we expect to be in the table? Is so, should it be `expectedItems`?

Reply:
its a test helper implementation of queryItems. At the beginning of the test, the table starts with whatever the initial state of the table is set in the test. The query items implementation returns what would happen as a result of the action. `Expected`  will be in the context of the test